### PR TITLE
Revamp crypto concept

### PR DESCRIPTION
**Pull request by segfaultxavi**
Add impact of stolen keys.

### CODE REVIEW

File: source/concepts/cryptography.rst [hunk 1]
```diff
@@ -2,97 +2,78 @@
...
 ********
 Key pair
 ********
 
-Elliptic curve cryptography is an approach to public key cryptography.
-The cryptographic system uses **pairs of keys**:
+Elliptic curve cryptography is based on **key pairs**: a private and a matching public key. In particular, |codename| uses the `Twisted Edwards curve <https://en.wikipedia.org/wiki/Twisted_Edwards_curve>`__ with the digital signature algorithm named `Ed25519 <https://ed25519.cr.yp.to>`__ and hashing algorithm **SHA-512**:
 
-* **Private key**: A random 256-bit integer used to sign :ref:`entities <verifiable-entity>` known by the owner.
+* **Private key**: A random 256-bit (64 byte) integer used to sign :ref:`entities <verifiable-entity>` known by the owner.
```
Comment (Jaguar0625) [inline source/concepts/cryptography.rst:15]: private keys are 32 bytes

File: source/concepts/cryptography.rst [hunk 2]
```diff
@@ -2,97 +2,78 @@
...
 
-* **Public key**: The public identifier of the key pair, which can be disseminated widely. It is used to prove that the entity was signed with the paired private key.
+* **Public key**: A 128-bit (32 bytes) integer derived from the private key. It serves as the public identifier of the key pair and can be disseminated widely. It is used to prove that an entity was signed with the paired private key.
 
-The public key is cryptographically derived from the private key.
-In particular, |codename| uses the |edwards| with the digital signature algorithm named |Ed25519| and hashing algorithm **SHA-512**.
+The public key can be derived from the private key, but **not the other way around**.
 
-You can find the |implementation-derivation| under the ``crypto`` module of :doc:`catapult-server <../server>`.
+The implementation can be found in the `crypto module <https://github.com/nemtech/catapult-server/blob/main/src/catapult/crypto>`__  of :doc:`catapult-server <../server>`.
```
Comment (Jaguar0625) [inline source/concepts/cryptography.rst:21]: crypto is just wrapper around external/donna

File: source/concepts/cryptography.rst [hunk 3]
```diff
@@ -2,97 +2,78 @@
...
+===============
 
-Public keys can be shared in a shorter form as addresses.
-A |codename| address is a **Base32 encoded triplet** consisting of:
+Key pairs are used in |codename| in different places, for **different purposes**. This is a summary of the keys used:
 
-* The network byte.
-* The 160-bit hash of the account's public key.
-* The 3 byte checksum, to allow the quick recognition of mistyped addresses.
+* **Main**: This key pair manages a **regular** :doc:`account`, containing mosaics.
```
Comment (Jaguar0625) [inline source/concepts/cryptography.rst:28]: mosaics => bit limited since it also gives ownership of other assets, like namespace,s mosaics etc

File: source/concepts/cryptography.rst [hunk 4]
```diff
@@ -2,97 +2,78 @@
...
-As you can see, it is possible to create an address without interacting with the blockchain.
-In fact, the blockchain only tracks addresses and public keys when they first appear in one transaction.
-
-.. |edwards| raw:: html
+*********
+Signature
+*********
 
-   <a href="https://en.wikipedia.org/wiki/Twisted_Edwards_curve" target="_blank">Twisted Edwards curve</a>
+**Messages can be signed** using a private key, producing 512-bit (128 byte) **signatures**.
```
Comment (Jaguar0625) [inline source/concepts/cryptography.rst:53]: 64 byte

File: source/concepts/cryptography.rst [hunk 5]
```diff
@@ -2,97 +2,78 @@
...
 
-.. |Ed25519| raw:: html
+Using the matching public key signatures can then **validate** that the key pair signed an entity.
```
Comment (Jaguar0625) [inline source/concepts/cryptography.rst:55]: missing a comma after public key?

File: source/concepts/cryptography.rst [hunk 6]
```diff
@@ -2,97 +2,78 @@
...
 
-   <a href="https://ed25519.cr.yp.to/" target="_blank">Ed25519</a>
+The implementation can be found in the `Signer class <https://github.com/nemtech/catapult-server/blob/main/src/catapult/crypto/Signer.cpp>`__ under the ``crypto`` module of :doc:`catapult-server <../server>`.
```
Comment (Jaguar0625) [inline source/concepts/cryptography.rst:57]: same comment about donna

**Review by Jaguar0625 — CHANGES_REQUESTED** [review]

File: source/concepts/cryptography.rst (hunk 2 above)
Comment (segfaultxavi) [reply source/concepts/cryptography.rst:21]: I prefer linking to the wrapper, so an interested reader can dig further. If we link directly to donna it's harder to get out and see who's calling it. No?

File: source/concepts/cryptography.rst (hunk 1 above)
Comment (segfaultxavi) [reply source/concepts/cryptography.rst]: 👍

File: source/concepts/cryptography.rst (hunk 3 above)
Comment (segfaultxavi) [reply source/concepts/cryptography.rst]: 👍

File: source/concepts/cryptography.rst (hunk 4 above)
Comment (segfaultxavi) [reply source/concepts/cryptography.rst]: 👍

File: source/concepts/cryptography.rst (hunk 5 above)
Comment (segfaultxavi) [reply source/concepts/cryptography.rst]: 👍

**Review by Jaguar0625 — APPROVED** [review]

File: source/concepts/cryptography.rst (hunk 2 above)
Comment (gimer) [reply source/concepts/cryptography.rst:21]: would skip the sentence completely, pubkey derivation is indeed in donna, and there's no much sense in pointing to wrapper, which simply calls `ed25519_publickey` (KeyPair.cpp)

File: source/concepts/cryptography.rst (hunk 6 above)
Comment (gimer) [reply source/concepts/cryptography.rst:57]: I somewhat agree. Signer could stay, as we're only using scalarmult and few others, but would reword it, as calling it `implementation` is a) overstatement b) it suggest it might be custom implementation, when it is not.

Comment (gimer) [reply source/concepts/cryptography.rst:57]: also, if you're mentioning this, it is weird, not to mention batched signature verification.

Comment (segfaultxavi) [reply source/concepts/cryptography.rst:57]: OK, I kept the links to the source because they were in the original version so I thought they must have brought value at some point. But the code structure is complex (with wrappers and all) and this is not the place to explain it so... I think we're better off without the code links. Interested readers will know where to look for the code.
